README.md

```diff
@@ -50,3 +50,6 @@ Test your new function; I've provided some basic testing code in `code.test.js`.
 What is the worst-case asymptotic time complexity of your implementation? What
 is the worst-case asymptotic memory complexity? Add your answer, including your
 reasoning, to this markdown file.
+I had a little help from ai to help understand the swapping and the stopping criteria.
+
+The worst case asymptotic time complexity of the implementation is O(n). The calculateRouteDistance function and the twoOptSwap function both operate in O(n) time and since each iteration of the algorithm involves these operations, a single iteration is O(n). The algorithm runs for a maximum of 100 iterations, which is based on my stoping criteria, which is a constant, so the overall time complexity remains O(n). the worst case asymptotic memory complexity is also O(n), as the algorithm stores the current route, which requires O(n) memory, and creates a new route during swaps, also requiring O(n) memory. Other variables require only O(1) space. So the algorithm is effecient in terms of both time and space.
```

Comment:
How many times could your loop run?

Reply:
worst case it would be 100 but i dont know how many times it does through each run if it found the solution in fewer runs